mailpoet/mailpoet
# Add "page view" tracking [MAILPOET-3897]

## Testing instructions
### Testing with cookie
1) Set tracking in MailPoet > Settings > Advanced to Full tracking
2) Send yourself an email with a link that goes to your site
3) Logout from the site and open the link (or use an anonymous window)
4) Check that cookie 'mailpoet_subscriber' is present (open console and click Application, then click the domain under Cookies and search for mentioned cookie
5) Reload the page
6) Check that cookie 'mailpoet_pave_view' is present (the same screen)
7) Go to admin and open stats for the subscriber who clicked the link and check that `Last engagement` (right under the heading) was updated

### Testing without cookie (for logged-in user)
1) Set tracking in MailPoet > Settings > Advanced to Partial tracking
2) Login to the site in an anonymous window and go to some front end page
3) Go to admin and open stats for the subscriber who clicked the link and check that `Last engagement` (right under the heading) was updated

### Check abandoned cart email
This change contains refactor of abandoned car rescheduling logic. Please check that abandoned cart emails work as expected. If the email is set to be sent 30 minutes after the last action it should be sent 30 minutes after the last subscriber's page visit. Please put any product to cart and wait for 30 mins in order to receive 'abandoned email' newsletter to your logged-in user's email address. Remember not to use testing site during that period.

[MAILPOET-3897]

[MAILPOET-3897]: https://mailpoet.atlassian.net/browse/MAILPOET-3897?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
